client/lib/signup/step-actions.js

```diff
@@ -487,32 +502,34 @@ export function createAccount(
 			( error, response ) => {
 				const errors =
 					error && error.error ? [ { error: error.error, message: error.message } ] : undefined;
+
+				if ( errors ) {
+					callback( errors );
```

Comment:
Yes, it's ok because in case of error, the `providedDependencies` param would go unused anyway. The callback dispatches the `processedSignupStep` action (renamed to `completeSignupStep`). If the action has the `errors` property, then [`invalidateStep` is performed and the rest is skipped](https://github.com/Automattic/wp-calypso/blob/ef6f2ed286d42f3a7b1fbe674317751acaa9641a/client/lib/signup/progress-store.js#L108-L111).